src/backend/bacon-video-widget-gst-missing-plugins.c

```diff
@@ -473,29 +473,92 @@ missing_codecs_supported (const char **codecs)
 	return TRUE;
 }
 
+typedef struct _CodecMissingSuggestedLaunchAction {
+	const gchar *message_text;
+	const gchar *button_text;
+	GDesktopAppInfo *desktop_info;
+	const gchar *mrl;
+} CodecMissingSuggestedLaunchAction;
+
+static CodecMissingSuggestedLaunchAction *
+codec_missing_suggested_launch_action_new (const gchar *message_text,
+					   const gchar *button_text,
+					   GDesktopAppInfo *desktop_info,
+					   const gchar *mrl)
+{
+	CodecMissingSuggestedLaunchAction *action = g_slice_new0 (CodecMissingSuggestedLaunchAction);
+	action->message_text = message_text;
+	action->button_text = button_text;
+	action->desktop_info = desktop_info;
+	action->mrl = mrl;
+
+	return action;
+}
+
+static void
+codec_missing_suggested_launch_action_free (CodecMissingSuggestedLaunchAction *action)
+{
+	g_object_unref (action->desktop_info);
+	g_slice_free (CodecMissingSuggestedLaunchAction, action);
+}
+
+static CodecMissingSuggestedLaunchAction *
+determine_vlc_launch_action_for_missing_codecs (const gchar *mrl) {
+	GDesktopAppInfo *candidate;
+	const gchar *message_text;
+
+	/* First try org.videolan.VLC.desktop, and if it exists
+	 * use that, otherwise use eos-vlc.desktop */
+	candidate = g_desktop_app_info_new ("org.videolan.VLC.desktop");
+	if (candidate) {
+		message_text = _("This file type is currently unsupported by the default "
```

Comment:
Don't mention `by the default video player.`, since the fact that it's unsupported is not related to whether or not Totem is the default player. (You could mention `in Totem`, or `in Videos`, but I'd just leave it without.)